src/software/geom/algorithms/rasterize.cpp

```diff
@@ -0,0 +1,293 @@
+#include "software/geom/algorithms/rasterize.h"
+
+#include <algorithm>
+#include <cmath>
+
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/algorithms/distance.h"
+#include "software/geom/algorithms/intersection.h"
+#include "software/geom/algorithms/rasterize.h"
+
+
+bool isInPixel(const Point& a, const Point& b, double resolution_size);
+std::optional<Point> isAVertex(const Point& point, const Polygon& polygon, double resolution_size);
+const Point& find_next_point_in_polygon(const Point& point, const Polygon& polygon);
+const Point& find_previous_point_in_polygon(const Point& point, const Polygon& polygon);
+bool is_special_vertex(const Point& point, const Polygon& polygon, const double resolution_size);
```

Comment:
1. "private" functions can go in rasterize_private.h/cpp files
2. nit: camelCase for function names

Reply:
These functions were helper functions and are not needed anymore, so we've removed them